### ISSUE TITLE
Discrepancy in AutoAttack Robustness Results 

### ISSUE THREAD

**Issue by TalRub104** [Biqing-Qi/Exploring-Adversarial-Robustness-of-Deep-State-Space-Models#7]
Hi,
I am using your adv_test function from utils/evaluation.py after adding AutoAttack to it (based on the AutoAttack implementation in train_trades_mnist.py). Below is the modified code:
`def adv_test(args, model, device, test_loader, log_path, attack_method = 'PGD'):
    model.eval()
    attack = None
    test_loss = 0
    correct = 0
    if attack_method == 'PGD':
        attack = PGD(model, args.epsilon, args.step_size, args.num_steps, normalize=False)
    elif attack_method =='APGD':
        attack = APGD(model, eps=args.epsilon, seed=args.seed,device=device)
    
    for data, target in test_loader:
        data, target = data.to(device), target.to(device)
        if attack_method =='APGD':
            data_adv = attack.perturb(data, target)
        elif attack_method == 'PGD':
            data_adv = attack(data, target)
        else:
            adversary = AutoAttack(model, norm=args.norm, eps=args.epsilon, log_path=log_path, version=args.version, verbose=args.AA_verbose)
            with torch.no_grad():
                if not args.individual:  # args.individual==False args.n_ex=2000
                    data_adv = adversary.run_standard_evaluation(data, target, bs=args.AA_bs, state_path=args.state_path)
                else:
                    # individual version, each attack is run on all test points
                    data_adv = adversary.run_standard_evaluation_individual(data, target, bs=args.batch_size)
        
        torch.cuda.empty_cache()
        with torch.no_grad():
            output = model(data_adv)
            test_loss += F.cross_entropy(output, target, size_average=False).item()
            pred = output.max(1, keepdim=True)[1]
            correct += pred.eq(target.view_as(pred)).sum().item()
            torch.cuda.empty_cache()

    test_loss /= len(test_loader.dataset)

    print('Adv Test: Average loss: {:.4f}, Adv Accuracy: {}/{} ({:.0f}%)'.format(test_loss, correct, len(test_loader.dataset),100. * correct / len(test_loader.dataset)))
    test_accuracy = 100. * correct / len(test_loader.dataset)
    return test_loss, test_accuracy`
    
After training your Mamba architecture on MNIST for 100 epochs using TRADES, I calculated the model's robustness to both PGD and AutoAttack using this 'adv_test' function.

For the PGD attack, I achieved an adversarial accuracy of 98.49, which is very close to the value reported in your paper.
For the AutoAttack, I observed an adversarial accuracy of 0, while your paper reports 52.85.

Could you please advise on what might be causing this discrepancy? I’d appreciate any suggestions or insights.
